Not able to compile leptonica due to lack of src/xtractprotos.c files

Dan,

I am not being able to compile leptonica since the removal of src/xtractprotos.c  from source (due to issue #119). 

make gives the follwoing output:
make[2]: *** No rule to make target `xtractprotos.o', needed by `xtractprotos'.  Stop.
make[2]: *** Waiting for unfinished jobs....
  CC     zlibmemstub.lo
make[2]: Leaving directory `/tmp/src/leptonica/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/tmp/src/leptonica'
make: *** [all] Error 2

Putting the file back from ./prog/xtractprotos.c fixed the error. Perhaps, to avoid the dupplication of the file, it should be enough to adapt the Makefile to compile from ../prog/xtractprotos.c

regards,
Guilherme

--- a/.gitignore
+++ b/.gitignore
@@ -22,7 +22,6 @@
 makefile
 Makefile
 Makefile.in
-xtractprotos
 /aclocal.m4
 /autom4te.cache
 /config/

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -73,7 +73,6 @@ pkginclude_HEADERS = allheaders.h alltypes.h                    \
  readbarcode.h recog.h regutils.h stack.h                       \
  stringcode.h sudoku.h watershed.h
 
-noinst_PROGRAMS = xtractprotos
 LDADD = liblept.la
 
 EXTRA_DIST = arrayaccess.h.vc                                   \
@@ -82,7 +81,8 @@ EXTRA_DIST = arrayaccess.h.vc                                   \
  morphtemplate1.txt morphtemplate2.txt                          \
  stringtemplate1.txt stringtemplate2.txt
 
-allheaders: $(liblept_la_SOURCES)
-	@test -x xtractprotos || echo "First run 'make xtractprotos'"
-	./xtractprotos -prestring=LEPT_DLL -protos=inline $(liblept_la_SOURCES)
+$(top_builddir)/prog/xtractprotos$(EXEEXT): liblept.la
+	$(MAKE) -C $(top_builddir)/prog xtractprotos$(EXEEXT)
 
+allheaders: $(top_builddir)/prog/xtractprotos$(EXEEXT) $(liblept_la_SOURCES)
+	cd $(srcdir) && $(abs_top_builddir)/prog/xtractprotos$(EXEEXT) -prestring=LEPT_DLL -protos=inline $(liblept_la_SOURCES)

